examples/channels/multi/multi.C

```diff
@@ -0,0 +1,51 @@
+#include "multi.h"
+#include <cstdlib>
+#include <ctime>
+#include <numeric>
+
+Main::Main(CkArgMsg *m) {
+  mainProxy = thisProxy;
+  numWorkers = CkNumPes();
+  CkPrintf("[MAIN] Creating channel, producer, and consumer.\n");
+  channel = CProxy_CkMultiChannel<2, int>::ckNew();
+  CProxy_JobConsumer consumer = CProxy_JobConsumer::ckNew(channel);
+  CProxy_JobProducer producer = CProxy_JobProducer::ckNew(channel);
+  // Run the consumers and producer
+  consumer.run();
+  producer.run();
+}
+
+void Main::stageI() {
+  CkPrintf("[MAIN] Setting a callback for data from any channel.\n");
+  CkCallback cb(CkIndex_Main::stageII(std::make_pair(0, 0)), mainProxy);
+  int channels[2];
+  std::iota(channels, channels + 2, 0);
+  channel.receiveFromAny(channels, 2, cb);
```

Comment:
Does this not block?

Reply:
Since it's requesting a callback and not receiving the value in place, it does not block. Perhaps `receiveFromAny` and `receive` when invoking a callback to distinguish them from a blocking call? `receiveViaCallback` gets a bit verbose when it comes to `receiveFromAnyChannelViaCallback`... any other suggestions for a name?

Reply:
Maybe `receiveAsync` and `receiveAnyChannelAsync`?